src/Features/Core/Portable/Navigation/INavigableItem.cs

```diff
@@ -33,7 +36,7 @@ internal interface INavigableItem
         /// </summary>
         bool IsImplicitlyDeclared { get; }
 
-        Document Document { get; }
+        NavigableDocument Document { get; }
```

Comment:
📝 This is the core of the change. Rather than holding a `Document` instance, we hold a `NavigableDocument` containing a subset of necessary information extracted from the `Document` and exposing a very similar API for consumers.

Reply:
nice.